jobs/periodic/deactivate-missing-coc/index.ts

```diff
@@ -0,0 +1,43 @@
+import {getLogger} from "log4js";
+import { prisma } from "../../../common/prisma";
+import validator from "validator";
+import toDate = validator.toDate;
+import {deactivateStudent} from "../../../common/student/activation";
+
+const logger = getLogger();
+
+export default async function execute() {
+    logger.info("Missing CoC deactivation job: looking for students wo have not supplied CoC within 8 Weeks after registration...");
+    await deactivateMissingCoc();
+}
+
+export async function deactivateMissingCoc() {
+    const today = new Date();
+    const eightWeeksAgoDate = new Date();
+    var eightWeeksAgo = today.getDate() - 56;
+    eightWeeksAgoDate.setDate(eightWeeksAgo);
+    const defaultingStudents = await prisma.student.findMany({
+        where: {
+            createdAt: {
+                gt: new Date("2021-11-01")
+            },
+            active: true,
+            screening: {
+                success: true,
+                createdAt: {
+                    lt: eightWeeksAgoDate
+                }
+            },
+            // eslint-disable-next-line camelcase
+            certificate_of_conduct: {
+                none: {}
+            }
+        }
+    });
+
+    logger.info(defaultingStudents.length+ " Defaulting students are ::: "+JSON.stringify(defaultingStudents));
```

Comment:
This is hard to track back from the logs, maybe Something like "5 students are going to be deactivated because of a missing CoC: "